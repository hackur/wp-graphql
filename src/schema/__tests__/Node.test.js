import { graphql } from 'graphql';
import { toGlobalId } from 'graphql-relay';
import { getMockSchema } from 'jest/utils';
import getLoaders from 'data/loaders';

const context = {};
const rootValue = {
  loaders: getLoaders(),
};

const queryByID = `
  query NodeQuery($id: ID!) {
    node(id: $id) {
      __typename
    }
  }
`;

describe('Test Node queries', () => {
  const schema = getMockSchema({
    Category: () => ({
      __typename: 'Category',
    }),
    Tag: () => ({
      __typename: 'Tag',
    }),
    Page: () => ({
      __typename: 'Page',
    }),
    Post: () => ({
      __typename: 'Post',
    }),
    User: () => ({
      __typename: 'User',
    }),
    Sidebar: () => ({
      __typename: 'Sidebar',
    }),
    NavMenu: () => ({
      __typename: 'NavMenu',
    }),
  });

  test('GraphQL should return a node by category ID', async () => {
    const variables = {
      id: toGlobalId('Category', 1),
    };
    const result = await graphql(schema, queryByID, rootValue, context, variables);
    expect(result).toMatchSnapshot();
  });

  test('GraphQL should return a node by tag ID', async () => {
    const variables = {
      id: toGlobalId('Tag', 1),
    };
    const result = await graphql(schema, queryByID, rootValue, context, variables);
    expect(result).toMatchSnapshot();
  });

  test('GraphQL should return a node by post ID', async () => {
    const variables = {
      id: toGlobalId('Post', 1),
    };
    const result = await graphql(schema, queryByID, rootValue, context, variables);
    expect(result).toMatchSnapshot();
  });

  test('GraphQL should return a node by page ID', async () => {
    const variables = {
      id: toGlobalId('Page', 1),
    };
    const result = await graphql(schema, queryByID, rootValue, context, variables);
    expect(result).toMatchSnapshot();
  });

  test('GraphQL should return a node by user ID', async () => {
    const variables = {
      id: toGlobalId('User', 1),
    };
    const result = await graphql(schema, queryByID, rootValue, context, variables);
    expect(result).toMatchSnapshot();
  });

  test('GraphQL should return a node by sidebar ID', async () => {
    const variables = {
      id: toGlobalId('Sidebar', 1),
    };
    const result = await graphql(schema, queryByID, rootValue, context, variables);
    expect(result).toMatchSnapshot();
  });

  test('GraphQL should return a node by NavMenu ID', async () => {
    const variables = {
      id: toGlobalId('NavMenu', 1),
    };
    const result = await graphql(schema, queryByID, rootValue, context, variables);
    expect(result).toMatchSnapshot();
  });
});
