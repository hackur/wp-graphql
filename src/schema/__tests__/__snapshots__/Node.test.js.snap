// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Node queries GraphQL should return a node by NavMenu ID 1`] = `
Object {
  "data": Object {
    "node": Object {
      "__typename": "Tag",
    },
  },
}
`;

exports[`Test Node queries GraphQL should return a node by category ID 1`] = `
Object {
  "data": Object {
    "node": Object {
      "__typename": "Sidebar",
    },
  },
}
`;

exports[`Test Node queries GraphQL should return a node by page ID 1`] = `
Object {
  "data": Object {
    "node": Object {
      "__typename": "Tag",
    },
  },
}
`;

exports[`Test Node queries GraphQL should return a node by post ID 1`] = `
Object {
  "data": Object {
    "node": Object {
      "__typename": "Category",
    },
  },
}
`;

exports[`Test Node queries GraphQL should return a node by sidebar ID 1`] = `
Object {
  "data": Object {
    "node": Object {
      "__typename": "Sidebar",
    },
  },
}
`;

exports[`Test Node queries GraphQL should return a node by tag ID 1`] = `
Object {
  "data": Object {
    "node": Object {
      "__typename": "Tag",
    },
  },
}
`;

exports[`Test Node queries GraphQL should return a node by user ID 1`] = `
Object {
  "data": Object {
    "node": Object {
      "__typename": "Category",
    },
  },
}
`;
